core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/AbstractHighlighterBuilder.java

```diff
@@ -119,10 +127,12 @@ protected AbstractHighlighterBuilder(StreamInput in) throws IOException {
         order(in.readOptionalWriteable(Order::readFromStream));
         highlightFilter(in.readOptionalBoolean());
         forceSource(in.readOptionalBoolean());
+        boundaryScannerType(in.readOptionalWriteable(BoundaryScannerType::readFromStream));
```

Comment:
Since this PR can go to 5.x can you add a check on the version, like this:
````` 
 if (in.getVersion().onOrAfter(Version.V_5_4_0_UNRELEASED)) { 
    boundaryScannerType(in.readOptionalWriteable(BoundaryScannerType::readFromStream));
}
`````
This will ensure that a node in 5.4 can safely send request to a 5.3 node.